### DISCUSSION TITLE
useBlock() hook

### DISCUSSION THREAD

**Discussion by brunowowk** [wevm/wagmi#2071] [Ideas]
Sometimes we need more than just the number from the latest block like the timestamp, hash, etc.

With the current `useBlockNumber()` hook, we only get the number.

I propose adding a `useBlock()` hook which includes [all block info](https://docs.ethers.org/v5/api/providers/types/#providers-Blockhttps://docs.ethers.org/v5/api/providers/types/#providers-Block)

**Comment by firnprotocol** [comment]
i have the same need. @jxom is this on the radar at all for v1? many thanks in advance.

**Reply by tmm** [reply]
All the actions supported by viem will eventually be reflected in wagmi. Once v1 is out, we'll work on adding support and contributions will be welcome!

**Reply by firnprotocol** [reply]
awesome! will be happy to look into raising a PR once v1 is released.

**Reply by firnprotocol** [reply]
any chance I could ask for an update on this? it's blocking my migration to wagmi. unfortunately, I've never written/used typescript, so it will be hard for me to contribute this directly. many thanks in advance.

**Reply by tmm** [reply]
We are working on some higher priority post-v1 items at the moment. In the meantime, it's trivial to get this data on your own:

```ts
import * as React from 'react'
import { usePublicClient } from 'wagmi'
 
function App() {
  const [block, setBlock] = React.useState()
  const publicClient = usePublicClient()

  React.useEffect(() => {
    publicClient
      .getBlock() // https://viem.sh/docs/actions/public/getBlock.html
      .then(x => setBlock(x))
      .catch(error => console.log(error))
  }, [publicClient])
}
```

**Reply by firnprotocol** [reply]
excellent!!! didn't realize it was this easy. consider this taken care of on my end; would be nice to have it officially but it's not urgent anymore.

**Comment by firnprotocol** [comment]
hi @jxom i hope all is well. i have to humbly beg you to include this feature in the V1 release. specifically, what I need are two `"wagmi/actions"`:
 1. `watchBlock`, analogous to [`watchBlockNumber`](https://wagmi.sh/core/actions/watchBlockNumber) in the obvious way; i.e., except now the callback is called on an entire block.
 2. `fetchBlock`, aka `fetchBlockByNumber`. here, you supply an optimal `blockNumber` argument, and async request the corresponding block. this isn't quite analogous to [`fetchBlockNumber`](https://wagmi.sh/core/actions/fetchBlockNumber), since that one appears to only return the most recent block.

for my usecase, i only need the result to contain the `timestamp` field. the rest can be added later in my view.

huge thanks in advance, and for all your work thus far.
